### ISSUE TITLE
BC 19.0 Artifacts => WebClient returns "500 - Internal server error"

### ISSUE THREAD

**Issue by gregoral** [microsoft/navcontainerhelper#2158]
A newly created  W1 BC container does not work.
WebClient returns: "500 - Internal server error"
![image](https://user-images.githubusercontent.com/42441688/139395146-02c6172f-1124-4589-bb3a-7a6a2554de6a.png)

**Steps taken:**
- create a new image from artifacts
- create a new container from image
- open web client

**Observations:**
- certificate download works fine
- bc server appears to be working ( Get-NAVAppInfo, Publish-NAVApp, Install-NAVApp )
- webclient simply returns error 500
- version 18.5 works fine
- version 19.0 does not
- windows error log does not appear to have anything unusual ( I can provide an evtx export if needed )
- using -isolation hyperv doesn't help

```
# 19.0 returns error 500 when opening webclient
New-BcImage -artifactUrl 'https://bcartifacts.azureedge.net/sandbox/19.0.29894.31767/w1'
docker run -d ... local-bc-image:sandbox-19.0.29894.31767-w1

# 18.5 works fine
New-BcImage -artifactUrl 'https://bcartifacts.azureedge.net/sandbox/18.5.29545.31744/w1'
docker run -d ... local-bc-image:sandbox-18.5.29545.31744-w1 
```

**BcContainerHelper command used to create the image**
```
New-BcImage `
   -artifactUrl 'https://bcartifacts.azureedge.net/sandbox/19.0.29894.31767/w1' `
   -imageName 'local-bc-image' `
   -isolation 'process' `
   -baseImage 'mcr.microsoft.com/dynamicsnav:10.0.17763.1577-generic' `
   -memory '8gb' `
   -skipDatabase:$false `
   -multitenant:$false `
   -addFontsFromPath 'C:\BcFonts' `
   -licenseFile 'C:\bc19.flf' `
   -includeTestToolkit:$true `
   -includeTestFrameworkOnly:$true `
   -includeTestLibrariesOnly:$true `
   -includePerformanceToolkit:$true
```

**Additional context**
```
BcContainerHelper is version 2.0.16
Using -isolation hyperv doesn't help.

local-bc-image:sandbox-19.0.29894.31767-w1
Building image local-bc-image:sandbox-19.0.29894.31767-w1 based on mcr.microsoft.com/dynamicsnav:10.0.17763.1577-generic with https://bcartifacts.azureedge.net/sandbox/19.0.29894.31767/w1
Generic Tag: 0.1.0.24
Container OS Version: 10.0.17763.1577 (ltsc2019)
Host OS Version: 10.0.17763.2237 (ltsc2019)
```

**Comment by freddydk** [comment]
Did you try to create a container using New-BcContainer?
If I am to help troubleshoot this, I would need the full output from creation of the Image and the full output from running the container.

**Comment by tpernus** [comment]
Hi,

I have the same problem. I was using New-BcContainer. I have updated docker verison 4.1.1 (69879). 
Here are the logs from container creation script.
[Container19_log_problem.txt](https://github.com/microsoft/navcontainerhelper/files/7497261/Container19_log_problem.txt)

**Comment by freddydk** [comment]
The base image 'mcr.microsoft.com/dynamicsnav:10.0.17763.1577-generic' is very very very very very very old.
In fact it doesn't exist anymore.
Some info here:
https://freddysblog.com/2020/12/06/moving-the-generic-images/

You should just remove the baseimage line in new-bcimage - then it should find the image that matches your OS the best.

**Comment by gregoral** [comment]
Hi,
we use a fixed base image because otherwise the build server keeps running out of disk space.

We have already tried to create an image letting BcContainerHelper choose the image freely.
It picked this one:
Pulling latest image mcr.microsoft.com/businesscentral:**10.0.17763.2237**
but the result is the same.

I have tried:
- creating with and without custom flf license file
- giving 24GB of memory to the container
- using 19.0, 19.1, 19.2 versions
- using New-BcImage and New-BcContainer

Here is one of the logs from image creation:
[New-BcImage.log.txt](https://github.com/microsoft/navcontainerhelper/files/7497526/New-BcImage.log.txt)
 
Older versions (18.6, 18.5, ...) work fine.

**Comment by freddydk** [comment]
Sorry, but I don't know what is happening here.
I can say that I have a lot of containers with 19.0 running in different places.
Also tried in a https://aka.ms/getbc container here: http://fkbc.westeurope.cloudapp.azure.com/
You could try to extract the event log and see if there is any information there.
If you create the container using new-bccontainer, then you can use Get-bcContainerEventlog to get the event log.

**Comment by gregoral** [comment]
We have managed to resolve the issue.
We're not 100% sure about the exact cause.

Uninstalling BcContainerHelper 1.0.7 helped to resolve the issue.
I'm not sure how this would be possible.
As far as I know only one version of a powershell module is active at any one time.
And we were using BcContainerHelper 2.0.16.

Idea came from a colleague that still had NavContainerHelper installed alongside BcContainerHelper.
Uninstalling NavContainerHelper solved the issue for her.

Uninstalling BcContainerHelper 1.0.7 solved the issue for me.

I don't know yet exactly what was different between these images.

I have noticed that this image uses **MSSQL15.SQLEXPRESS**.
I'm quite sure this is different because restore of my backup failed on the working image and it was not failing before.

On the latest image RelocateFile flag was needed when restoring data from v18 backup file.

I'm not sure if this is the root of the issue but it is definitely something that changed between these images.

Hope this helps someone in the future. Because this issue was really elusive.
We're still not 100% sure if an old version of BcContainerHelper was the issue.
It might be that uninstalling the old BcContainerHelper version just caused some other change that was the cause.

BC server appeared to work fine the whole time.
Even on the "bad" images database and app upgrades went trough without any issues.
And the database that I got out work on the latest image/container without any issues .
At least I have not noticed anything so far.
